x/mongo/driver/operation.go

```diff
@@ -764,6 +767,12 @@ func (op Operation) createQueryWireMessage(dst []byte, desc description.Selected
 		return dst, info, err
 	}
 
+	// Add server API information before calling addSession() because it will modify
+	// transaction state.
+	if op.Client == nil || !op.Client.TransactionInProgress() {
```

Comment:
Transactions that have been committed can be repeatedly committed:
https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#clientsession-changes

The state of the session after the first commit changes to `Committed`. 

I think there could be a danger when double committing a transaction:
- On the first commit, `TransactionInProgress() == true`, no server API fields are appended.
- On the second commit, `TransactionInProgress() == false` returns false on the second commit. So this block is entered.

Updating transaction-handling.json to append an extra commit seems to still pass though. So I may be misunderstanding.

Reply:
I agree! I'll bring it up in the versioned-api implementation channel before submitting a PR but will link it in this thread if/when I end up making one.